examples/cpp/pyperf/PyPerfBPFProgram.cc

```diff
@@ -473,14 +475,24 @@ get_first_arg_name(
   char *argname,
   size_t maxlen) {
   int result = 0;
+  ssize_t ob_size; // Py_ssize_t;
   // Roughly equivalnt to the following in GDB:
   //
   //   ((PyTupleObject*)$frame->f_code->co_varnames)->ob_item[0]
   //
   void* args_ptr;
   result |= bpf_probe_read_user(&args_ptr, sizeof(void*), code_ptr + offsets->PyCodeObject.co_varnames);
-  result |= bpf_probe_read_user(&args_ptr, sizeof(void*), args_ptr + offsets->PyTupleObject.ob_item);
-  result |= bpf_probe_read_user_str(argname, maxlen, args_ptr + offsets->String.data);
+  result |= bpf_probe_read_user(&ob_size, sizeof(ob_size), args_ptr + offsets->String.size); // String.size is PyVarObject.ob_size
+  if (ob_size > 0) {
```

Comment:
Also need `result != 0` here.

Reply:
`bpf_probe_read_user` clears the buffer on error (so `ob_size` will be `0` if this call fails). If the previous call fails, then we are reading from `0 + offsets->String.size` which will fail as well (NULL page read, ...).

But for correctness, okay

Reply:
Hmm, touché 👍